interfaces/builtin/ros_opt_data.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const rosOptDataSummary = `Allows read-only access to the static data such as xacro,yaml,urdf,stl,... in the standard /opt/ros folder`
+
+const rosOptDataBaseDeclarationSlots = `
+  ros-opt-data:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: false
+`
+
+const rosOptDataConnectedPlugAppArmor = `
+# Description: Allows read-only access to the xacro,yaml,urdf,stl files in /opt/ros folder
+# capability dac_read_search,  # this should not be necessary, the idea is to read only what normal user can read
+/var/lib/snapd/hostfs/opt/ros/ r,
```

Comment:
is the fact that they are not at /opt/ros a big annoyance or not? I suppose not given you wrote this way, but wondering because in principle the interface could also maybe mount things there inside the snap namespace?

Reply:
Indeed I also considered mounting them in the same /opt/ros convenient folder as the host but I felt it would be confusing for some kind of usages. 
For example if the snap using this interface is already putting something else in /opt/ros and would see something overwritten or shadowed. 
Another example is if the Ros version in the host is not fully compatible with the Ros version inside the snap -> the default folder would be loaded automatically with broken data, while a separate folder can be managed with additional checks. This is mostly true for *.so libraries which I am filtering anyway, but not only for libraries.